libs/ui/switch/brain/src/lib/brn-switch.component.ts

```diff
@@ -152,13 +153,14 @@ export class BrnSwitchComponent implements AfterContentInit, OnDestroy {
 	@ViewChild('checkBox', { static: true })
 	public checkbox?: ElementRef<HTMLInputElement>;
 
-	@Output()
-	public changed = new EventEmitter<boolean>();
-	@Output()
-	public touched = new EventEmitter<void>();
+	public readonly changed = output<boolean>();
+	public readonly touched = output<void>();
+	private readonly destroyRef = inject(DestroyRef);
```

Comment:
Is this needed here? I thought takeUntilDestroyed takes care of this automatically when used in the constructor? I might be wrong though

Reply:
There were error message, that said that we are subscribing on already destroyed component.

Reply:
This sounds strange, should not be necessary to add the destroyRef.
Maybe there was another issue with your code?

Reply:
@goetzrobin was right. Argument is not needed. However unsubscribe was missing ...